### PR TITLE
DBZ-1078 Cache sys.fn_cdc_map_lsn_to_time() results

### PR DESCRIPTION
**Pull request by grzegorz8**

### CODE REVIEW

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java [hunk 1]
```diff
@@ -70,6 +74,7 @@
      */
     public SqlServerConnection(Configuration config) {
         super(config, FACTORY);
+        lsnToInstantCache = new BoundedConcurrentHashMap<>(100);
```
Comment (grzegorz8) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java]: Is it ok to have a hardcoded cache size?

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java [hunk 2]
```diff
@@ -164,12 +169,19 @@ public Instant timestampOfLsn(Lsn lsn) throws SQLException {
...
+            return cachedInstant;
+        }
+
         return prepareQueryAndMap(query, statement -> {
             statement.setBytes(1, lsn.getBinary());
         }, singleResultMapper(rs -> {
             final Timestamp ts = rs.getTimestamp(1);
             final Instant ret = (ts == null) ? null : ts.toInstant();
             LOGGER.trace("Timestamp of lsn {} is {}", lsn, ret);
+            Optional.ofNullable(ret).ifPresent(v -> lsnToInstantCache.put(lsn, ret));
```
Comment (jchipmunk) [inline debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java]: usage of `Optional` here - is an extra memory allocation, let's replace it to `if` condition

Comment (grzegorz8) [reply debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java]: Done.

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java (hunk 1 above)
Comment (gunnarmorling) [reply debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnection.java]: Yes, I think that's fine.

**Comment by gunnarmorling** [comment]
LGTM, merging. Thanks, @grzegorz8!

**Comment by gunnarmorling** [comment]
And thanks for reviewing, @jchipmunk!

**Pull request merged**